### PR TITLE
Add aws sdk s3ForcePathStyle option

### PR DESCRIPTION
**Pull request by nika-begiashvili**
Some non aws S3 implementations do not serve objects at bucket.region.endpoint location but rather use endpoint/bucket style. to facilitate this kind of behavior aws-sdk-js provides `s3ForcePathStyle` option.
this pull request exposes that option to cdn config

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -219,6 +219,12 @@ const schema = {
         format: String,
         default: '',
         env: 'AWS_S3_IMAGES_ENDPOINT'
+      },
+      pathStyle: {
+        doc: 'Forces path style access to s3 storage, helpful for storage systems with S3 support. Not required for Amazon S3 and Digital Ocean Spaces.',
```
Comment (eduardoboucas) [inline config.js]:
```suggestion
        doc: 'Force path-style URLs when retrieving images from S3 (https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html#access-bucket-intro)',
```

File: config.js [hunk 2]
```diff
@@ -292,6 +298,12 @@ const schema = {
         format: String,
         default: '',
         env: 'AWS_S3_ASSETS_ENDPOINT'
+      },
+      pathStyle: {
+        doc: 'Forces path style access to s3 storage, helpful for storage systems with S3 support. Not required for Amazon S3 and Digital Ocean Spaces.',
```
Comment (eduardoboucas) [inline config.js]:
```suggestion
        doc: 'Force path-style URLs when retrieving assets from S3 (https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html#access-bucket-intro)',
```

**Comment by jimlambie** [comment]
@nika-begiashvili I'm adding a unit test to the application to prove this change does what we expect it to. Are you able to provide me with a sample configuration?

**Comment by jimlambie** [comment]
@nika-begiashvili In addition to the above request for sample configuration details, can you give us any more details about the S3-like service you are using? You mention "some non-AWS S3 implementations" (of which Digital Ocean Spaces is one), but in order for us to ensure we support your requirement we will need to test against the same service you wish to use.

cc @eduardoboucas

**Comment by nika-begiashvili** [comment]
@jimlambie thank you for support, but starting to doubt this PR, since this is deprecated way to use S3 I believe better option would be to rewrite url using reverse proxy
the storage I'm using is caringo swarm which unfortunately is not open-source, maybe fake-s3 or pithos has path style access but I'm not sure

**Comment by nika-begiashvili** [comment]
I've managed to make it work without this PR, using reverse proxy to rewrite urls
I'm closing this PR

**Comment by eduardoboucas** [comment]
@nika-begiashvili thanks for the update. And do let us know if there's anything we can help you with in the future!

**Pull request closed without merging**